python-package/xgboost/core.py

```diff
@@ -621,6 +638,19 @@ def set_float_info_npy2d(self, field, data):
                                                c_data,
                                                c_bst_ulong(len(data))))
 
+    def set_cudf_info(self, field, data):
+        col_ptrs = []
+        if isinstance(data, CUDF):
+            col_ptrs = [data[col]._column.cffi_view for col in data.columns]
+        else:
+            # data is a single CUDF column
+            col_ptrs = [data.cffi_view]
+        col_ptr_arr = CUDF_FFI.new('gdf_column*[]', col_ptrs)
+        _check_call(_LIB.XGDMatrixSetCUDFInfo
+                    (self.handle, c_str(field),
+                     ctypes.c_void_p(int(CUDF_FFI.cast('uintptr_t', col_ptr_arr))),
```

Comment:
Likewise, this call uses the FFI to cast column data into another type. This will not work if `cudf` is not installed.